### PR TITLE
Preserve recipients in MessageCompose during rotation (activity re-creation)

### PR DESCRIPTION
**Pull request by dabbler011**
Fixes: #3049

### CODE REVIEW

File: k9mail/src/main/java/com/fsck/k9/view/RecipientSelectView.java [hunk 1]
```diff
@@ -287,6 +287,16 @@ public void addRecipients(Recipient... recipients) {
         }
     }
 
+    public List<Recipient> getRecipients() {
+        return getObjects();
+    }
+
+    public void setRecipients(List<Recipient> recipients) {
```
Comment (philipwhiuk) [inline k9mail/src/main/java/com/fsck/k9/view/RecipientSelectView.java]: This should either clear all objects first or be renamed `addRecipients`

File: k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java [hunk 2]
```diff
@@ -142,6 +144,9 @@
     private static final String STATE_KEY_READ_RECEIPT = "com.fsck.k9.activity.MessageCompose.messageReadReceipt";
     private static final String STATE_KEY_CHANGES_MADE_SINCE_LAST_SAVE = "com.fsck.k9.activity.MessageCompose.changesMadeSinceLastSave";
     private static final String STATE_ALREADY_NOTIFIED_USER_OF_EMPTY_SUBJECT = "alreadyNotifiedUserOfEmptySubject";
+    private static final String TO_STRING = "to_string";
+    private static final String CC_STRING = "cc_string";
+    private static final String BCC_STRING = "bcc_string";
```
Comment (philipwhiuk) [inline k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java]: `STATE_TO_RECIPIENTS`

**Comment by dabbler011** [comment]
@philipwhiuk updated the commit. Below is the GIF of updates.

File: k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java [hunk 3]
```diff
@@ -142,6 +144,9 @@
     private static final String STATE_KEY_READ_RECEIPT = "com.fsck.k9.activity.MessageCompose.messageReadReceipt";
     private static final String STATE_KEY_CHANGES_MADE_SINCE_LAST_SAVE = "com.fsck.k9.activity.MessageCompose.changesMadeSinceLastSave";
     private static final String STATE_ALREADY_NOTIFIED_USER_OF_EMPTY_SUBJECT = "alreadyNotifiedUserOfEmptySubject";
+    private static final String STATE_TO_RECIPIENTS = "to_string";
```
Comment (philipwhiuk) [inline k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java]: Still called "to_string" - match the format of the line above

File: k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java [hunk 4]
```diff
@@ -596,6 +608,9 @@ public void onPause() {
     protected void onSaveInstanceState(Bundle outState) {
         super.onSaveInstanceState(outState);
 
+        outState.putString(STATE_TO_RECIPIENTS,toView.getText().toString().replace(",","").replace(" ",""));
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java]: Why remove all commas and spaces?

File: k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java [hunk 5]
```diff
@@ -625,6 +640,42 @@ protected void onRestoreInstanceState(Bundle savedInstanceState) {
...
+        List<RecipientSelectView.Recipient> ccRecipients = ccView.getRecipients();
+        List<RecipientSelectView.Recipient> bccRecipients = bccView.getRecipients();
+
+        toView.removeRecipients(toRecipients);
+        ccView.removeRecipients(ccRecipients);
+        bccView.removeRecipients(bccRecipients);
+
+        String toText = savedInstanceState.getString(STATE_TO_RECIPIENTS);
+        if(toText.length() != 0) {
+            if (toText.charAt(0) == '+') {
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java]: What's special about `+`?

**Pull request closed without merging**